tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommandOptions.java

```diff
@@ -0,0 +1,257 @@
+                    "Option " + describeOpt + " takes one of these options: " + join(allGroupSelectionScopeOpts, ", "));
+            List<OptionSpec<?>> mutuallyExclusiveOpts = Arrays.asList(membersOpt, offsetsOpt, stateOpt);
+            if (mutuallyExclusiveOpts.stream().mapToInt(o -> options.has(o) ? 1 : 0).sum() > 1) {
+                CommandLineUtils.printUsageAndExit(parser,
+                    "Option " + describeOpt + " takes at most one of these options: " + join(mutuallyExclusiveOpts, ", "));
+            }
+            if (options.has(stateOpt) && options.valueOf(stateOpt) != null)
+                CommandLineUtils.printUsageAndExit(parser,
+                    "Option " + describeOpt + " does not take a value for " + stateOpt);
+        } else {
+            if (options.has(timeoutMsOpt))
+                LOGGER.debug("Option " + timeoutMsOpt + " is applicable only when " + describeOpt + " is used.");
+        }
+
+        if (options.has(deleteOpt)) {
+            if (!options.has(groupOpt) && !options.has(allGroupsOpt))
+                CommandLineUtils.printUsageAndExit(parser,
+                    "Option " + deleteOpt + " takes one of these options: " + join(allGroupSelectionScopeOpts, ", "));
+            if (options.has(topicOpt))
+                CommandLineUtils.printUsageAndExit(parser, "The consumer does not support topic-specific offset " +
+                    "deletion from a consumer group.");
+        }
+
+        if (options.has(deleteOffsetsOpt)) {
+            if (!options.has(groupOpt) || !options.has(topicOpt))
+                CommandLineUtils.printUsageAndExit(parser,
+                    "Option " + deleteOffsetsOpt + " takes the following options: " + join(allDeleteOffsetsOpts, ", "));
+        }
+
+        if (options.has(resetOffsetsOpt)) {
+            if (options.has(dryRunOpt) && options.has(executeOpt))
+                CommandLineUtils.printUsageAndExit(parser, "Option " + resetOffsetsOpt + " only accepts one of " + executeOpt + " and " + dryRunOpt);
+
+            if (!options.has(dryRunOpt) && !options.has(executeOpt)) {
+                System.err.println("WARN: No action will be performed as the --execute option is missing." +
+                    "In a future major release, the default behavior of this command will be to prompt the user before " +
+                    "executing the reset rather than doing a dry run. You should add the --dry-run option explicitly " +
+                    "if you are scripting this command and want to keep the current default behavior without prompting.");
+            }
+
+            if (!options.has(groupOpt) && !options.has(allGroupsOpt))
+                CommandLineUtils.printUsageAndExit(parser,
+                    "Option " + resetOffsetsOpt + " takes one of these options: " + join(allGroupSelectionScopeOpts, ", "));
+            CommandLineUtils.checkInvalidArgs(parser, options, resetToOffsetOpt,   minus(allResetOffsetScenarioOpts, resetToOffsetOpt));
```

Comment:
hmm -- do we want to line up all the minus arguments here? It adds extra spaces that may be against our style guidelines

Reply:
Extra spaces removed